app/services/layout/index.ts

```diff
@@ -56,6 +56,42 @@ class LayoutViews extends ViewHandler<ILayoutServiceState> {
   className(layout: ELayout) {
     return LAYOUT_DATA[layout].className;
   }
+
+  calculateColumnTotal(slots: IVec2Array) {
+    let totalWidth = 0;
+    slots.forEach(slot => {
+      if (Array.isArray(slot)) {
+        totalWidth += this.calculateMinimum('x', slot);
+      } else if (slot) {
+        totalWidth += slot.x;
+      }
+    });
+
+    return totalWidth;
+  }
+
+  calculateMinimum(orientation: 'x' | 'y', slots: IVec2Array) {
+    const aggregateMins: number[] = [];
+    const minimums = [];
+    slots.forEach(slot => {
+      if (Array.isArray(slot)) {
+        aggregateMins.push(this.aggregateMinimum(orientation, slot));
+      } else {
+        minimums.push(slot[orientation]);
+      }
+    });
+    if (!minimums.length) minimums.push(10);
+    return Math.max(...minimums, ...aggregateMins);
+  }
+
+  aggregateMinimum(orientation: 'x' | 'y', slots: IVec2Array) {
+    const minimums = slots.map(mins => {
+      if (mins) return mins[orientation];
+      return 10;
+    });
+    if (!minimums.length) minimums.push(10);
+    return minimums.reduce((a: number, b: number) => a + b);
+  }
```

Comment:
Nice.  I didn't really anticipate views being used for shared logic that isn't based on state (i.e. pure helper functions) but I think it's a fine pattern.  Another pattern (which we don't utilize enough) is just exporting a function from a services file.  If it's just a pure function that doesn't interact with any state at all, it's totally fine to just define the function outside the service and export it.